_includes/header.html

```diff
@@ -17,73 +17,152 @@
+            <div class="landing-page-redesign--nav-search--field-with-results--field">
+              <div class="landing-page-redesign--nav-search--field-with-results--field--wrapper">
+                <input type="text" id="search-input" class="search-input"
+                  placeholder="Search for Anything" aria-label="Search {{ site.title }}"
+                  data-docs-version="latest" autocomplete="off"
+                >
+                <div class="search-spinner"><i></i></div>
+                <label for="search-input" class="search-label">
+                  <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="search-icon" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search">
+                    <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line>
+                  </svg>
+                </label>
+              </div>
+              <div id="search-results" class="search-results custom-search-results"></div>
+            </div>
+          </div>
+          <div class="search-overlay"></div>
+        </li>
       </ul>
     </div>
   </div>
 </div>
+<script type="module">
+  document.addEventListener('DOMContentLoaded', () => {
+    const subMenuHeights = [
+      '336px',
+      '144px',
+      '288px',
+      '240px',
+    ];
+    function getSubMenu(button) {
+      const parentLI = button.closest('li');
+      const childUL = parentLI.querySelector('ul');
+      return childUL;
+    }
+    function initializeCustomMenuHeights(button, index) {
+        const height = subMenuHeights[index];
+        const childUL = getSubMenu(button);
+        childUL?.style?.setProperty?.('--expanded-height', height);
+    }
+    function onToggleButtonClick(e) {
+      const visibleClassName = 'opensearch-toggle-button-link__visible';
+      const visibleSelector = `.${visibleClassName}`;
+      const invisibleClassName = 'opensearch-toggle-button-link__invisible';
+      const invisibleSelector = `.${invisibleClassName}`;
+
+      const toggle = e.currentTarget;
+      const visibleLink = toggle.querySelector(visibleSelector);
+      const invisibleLink = toggle.querySelector(invisibleSelector);
+      visibleLink.classList.remove(visibleClassName);
+      visibleLink.classList.add(invisibleClassName);
+      invisibleLink.classList.remove(invisibleClassName);
+      invisibleLink.classList.add(visibleClassName);
+      
+      const childUL = getSubMenu(toggle);
+      childUL?.toggleAttribute?.('expanded');
+    }
+    const topNavigationToggleButtons = document.querySelectorAll('#top .opensearch-toggle-button--wrapper');
+    for (let i = 0; i < topNavigationToggleButtons.length; ++i) {
+      const button = topNavigationToggleButtons[i];
+      initializeCustomMenuHeights(button, i);
+      button.addEventListener('click', onToggleButtonClick);
```

Comment:
One delegate should handle all of these; adding too many event listeners could slow the page on certain platforms.

Reply:
Agreed. The event listener is one static function reference, and there are only 4 buttons, and I have hardcoded the submenu heights for the custom CSS property to be applied as `--expanded-height`. I did this quick and dirty to get it done, however I have this resolution prioritized as a fast follow; along with a runtime determination of the height. This is for being able to do a smooth height transition.